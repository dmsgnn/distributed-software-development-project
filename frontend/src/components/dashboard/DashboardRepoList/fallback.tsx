import { Flex, FlexProps, Grid, GridProps, Skeleton } from '@chakra-ui/react';
import { FC } from 'react';

export const DashboardRepoListItemFallback: FC<FlexProps> = (props) => (
	<Flex direction="column" gap={2} {...props}>
		<Skeleton w="full" flex={6} borderRadius="lg" />
	</Flex>
);

export const DashboardRepoListFallback: FC<GridProps> = (props) => {
	return (
		<Grid templateColumns="repeat(4, 1fr)" gap={12} {...props}>
			<DashboardRepoListItemFallback />
			<DashboardRepoListItemFallback />
			<DashboardRepoListItemFallback />
			<DashboardRepoListItemFallback />
		</Grid>
	);
};
