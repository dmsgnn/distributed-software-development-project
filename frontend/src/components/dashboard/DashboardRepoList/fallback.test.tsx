import { axe } from 'jest-axe';

import { act, render } from 'test-utils';
import { DashboardRepoListFallback, DashboardRepoListItemFallback } from './fallback';

const mockRouterPush = jest.fn();
jest.mock('next/router', () => ({ useRouter: jest.fn(() => ({ push: mockRouterPush })) }));

describe('DashboardRepoListFallback', () => {
	it('should matches snapshot', () => {
		const { asFragment } = render(<DashboardRepoListFallback />);

		expect(asFragment()).toMatchSnapshot();
	});

	it('should matches snapshot DashboardRepoListItemFallback', () => {
		const { asFragment } = render(<DashboardRepoListItemFallback />);

		expect(asFragment()).toMatchSnapshot();
	});

	it('should be accessible', async () => {
		const { container } = render(<DashboardRepoListFallback />);

		let results: Awaited<ReturnType<typeof axe>>;
		await act(async () => {
			results = await axe(container);
		});

		expect(results).toHaveNoViolations();
	});
});
