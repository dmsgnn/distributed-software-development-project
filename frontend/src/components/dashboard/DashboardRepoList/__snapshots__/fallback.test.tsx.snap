// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardRepoListFallback should matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-bl1qof"
  >
    <div
      class="css-3a3hw0"
    >
      <div
        class="chakra-skeleton css-t9zdla"
      />
    </div>
    <div
      class="css-3a3hw0"
    >
      <div
        class="chakra-skeleton css-t9zdla"
      />
    </div>
    <div
      class="css-3a3hw0"
    >
      <div
        class="chakra-skeleton css-t9zdla"
      />
    </div>
    <div
      class="css-3a3hw0"
    >
      <div
        class="chakra-skeleton css-t9zdla"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`DashboardRepoListFallback should matches snapshot DashboardRepoListItemFallback 1`] = `
<DocumentFragment>
  <div
    class="css-3a3hw0"
  >
    <div
      class="chakra-skeleton css-t9zdla"
    />
  </div>
</DocumentFragment>
`;
